### PR TITLE
fix: gutenberg-based blocks with innerblocks

### PR DESCRIPTION
**Pull request by khawkins98**
This gets react-based innerblocks working again. This _mostly_ works, but there's something happening I've not yet been able to nail down — if you add more than one vf-grid to the page the observers start to go wild.

So far this PR only addresses vf-grid.

- drops the legacy `ExperimentalBlock` for `useBlockProps`
- `renderAppender` cannot be undefined, instead deafults to `InnerBlocks.DefaultBlockAppender`

related: https://github.com/WordPress/gutenberg/tree/trunk/packages/block-editor/src/components/inner-blocks

for: https://gitlab.ebi.ac.uk/emblorg/backlog/-/issues/480

I've moved from `__experimentalBlock` to `useBlockProps`, but I'm not sure this is necesary for this fix.

### CODE REVIEW

File: wp-content/plugins/vf-gutenberg/blocks/vf-core/vf-grid-column.jsx [hunk 1]
```diff
@@ -126,16 +126,16 @@ settings.edit = (props) => {
...
         </InspectorControls>
       )}
-      <ExperimentalBlock.div {...rootAttr}>
+      <div { ...blockProps }>
         <InnerBlocks
           templateLock={false}
           renderAppender={
             hasChildBlocks
-              ? undefined
+              ? InnerBlocks.DefaultBlockAppender
```
Comment (khawkins98) [inline wp-content/plugins/vf-gutenberg/blocks/vf-core/vf-grid-column.jsx:134]: This is probably the most important change.

**Comment by khawkins98** [comment]
Thinking about this more, I think there's actually a second bug at play somewhere else in  vf-wp's jsx.

**Comment by khawkins98** [comment]
Probably this 

https://github.com/visual-framework/vf-wp/blob/5b18e9f99bc4c92cba914970a6ab12f4f8518369/wp-content/plugins/vf-gutenberg/blocks/vf-core/vf-plugin.jsx#L57-L60

**Comment by sandykadam** [comment]
@nitinja do you have any thoughts on this? since its react.

**Comment by nitinja** [comment]
I am trying to setup this project on my windows machine (or may be virtualboxed ubuntu).

**Comment by khawkins98** [comment]
I can shed a bit more light on what the remaining issue is here. 

In `vf-grid.jsx` we need to set some styles so the Gutenberg editor shows the columns in a column layout:

```js
  const className = `vf-grid | vf-grid__col-${columns}`;
  const styles = {
    ['--block-columns']: columns
  };
  props.setAttributes({className: className})
  props.setAttributes({style: styles})
```

However, setting classes like this (although it seems to be the correct way) spawns massive loops — the component is invoked about 100 times a second. This eventually leads to React bailing.

image

With the above code disabled, things work fine however you don't get the correct editor layout.

image

**Review by nitinja — APPROVED** [review]

**Comment by kasprzyk-sz** [comment]
I revert this PR as it was merged to `master` while still being WIP

**Pull request merged**